libs/dal/src/types/enforce.ts

```diff
@@ -0,0 +1,5 @@
+import type { EnvironmentId } from '../repositories/environment';
+import type { OrganizationId } from '../repositories/organization';
+
+export type EnforceOrgId = { _organizationId: OrganizationId };
+export type EnforceEnvOrOrgIds = { _environmentId: EnvironmentId } | EnforceOrgId;
```

Comment:
Great! Only still missing thing (related to what I have pending) is that you could pass any string of any type and it will pass the validation, but for the enforcement query right now is enough. We can make it stricter in the future. 💪🏻